### PR TITLE
Refactor analysis output metrics, part 9: tune using perform_utility_analysis_new

### PR DESCRIPTION
**Pull request by dvadym**
This PR implements `tune_new` which uses `perform_utility_analysis_new` and returns new the result in new dataclasses.

Note: after migrating old code, the old `tune` will be dropped and `tune_new` will be renamed to `tune`

### CODE REVIEW

File: analysis/parameter_tuning.py [hunk 1]
```diff
@@ -252,6 +261,118 @@ def tune(col,
...
+        utility_result = result
+    # utility_result: (UtilityReport,)
+    # per_partition_utility_result: (pk, (PerPartitionMetrics))
+    use_public_partitions = public_partitions is not None
+
+    utility_result = backend.to_list(utility_result, "To list")
+    # 1 element collection with list[UtilityReport]
+    utility_result = backend.map(
+        utility_result,
+        lambda result: _convert_utility_analysis_to_tune_result_new(
```
Comment (dvadym) [inline analysis/parameter_tuning.py:336]: the difference with tune, only 
1.` backend.to_list`
2. and `_convert_utility_analysis_to_tune_result` -> `_convert_utility_analysis_to_tune_result_new`

File: analysis/parameter_tuning.py [hunk 2]
```diff
@@ -175,10 +176,18 @@ def _convert_utility_analysis_to_tune_result(
...
-                      index_best, utility_analysis_result)
+    return TuneResult(tune_options,
+                      contribution_histograms,
+                      run_configurations,
+                      index_best,
+                      utility_analysis_result,
+                      utility_reports=[])
 
 
+############# Deprected code #################
```
Comment (miracvbasaran) [inline analysis/parameter_tuning.py:187]: nit: `Deprecated`

File: analysis/parameter_tuning.py (hunk 1 above)
Comment (miracvbasaran) [reply analysis/parameter_tuning.py:336]: Why do we need the `backed.to_list`?

File: analysis/parameter_tuning.py [hunk 3]
```diff
@@ -252,6 +261,118 @@ def tune(col,
...
+    # TODO(dvadym): implement relative error.
+    # TODO(dvadym): take into consideration partition selection from private
+    # partition selection.
+    assert tune_options.function_to_minimize == MinimizingFunction.ABSOLUTE_ERROR
+
+    # Sort utility reports by configuration index.
+    utility_reports.sort(key=lambda e: e.configuration_index)
+
+    index_best = -1  # not found
+    # Find best index if there are metrics to compute. Absence metrics to
```
Comment (miracvbasaran) [inline analysis/parameter_tuning.py:360]: nit: `Absence of metrics to compute`?

**Review by miracvbasaran — APPROVED** [review]

**Review by dvadym — COMMENTED** [review]
Thanks for review!

**Pull request merged**